pypit/par/pypitpar.py

```diff
@@ -0,0 +1,2486 @@
+        kwargs = {}
+        for pk in parkeys:
+            kwargs[pk] = cfg[pk] if pk in k else None
+
+        pk = 'pca'
+        kwargs[pk] = PCAPar.from_dict(cfg[pk]) if pk in k else None
+
+        return cls(**kwargs)
+
+    @staticmethod
+    def valid_functions():
+        """
+        Return the list of valid functions to use for slit tracing.
+        """
+        return [ 'polynomial', 'legendre', 'chebyshev' ]
+
+    @staticmethod
+    def valid_sobel_modes():
+        """Return the valid sobel modes."""
+        return [ 'nearest', 'constant' ]
+
+    def validate(self):
+        if self.data['number'] == 0:
+            raise ValueError('Number of slits must be -1 for automatic identification or '
+                             'greater than 0')
+        
+class TraceTiltsPar(ParSet):
+    """
+    Parameters specific to PypIts slit tracing algorithm
+    """
+    def __init__(self, idsonly=None, method=None, params=None, order=None, disporder=None):
+
+        # Grab the parameter names and values from the function
+        # arguments
+        args, _, _, values = inspect.getargvalues(inspect.currentframe())
+        pars = dict([(k,values[k]) for k in args[1:]])      # "1:" to skip 'self'
+
+        # Initialize the other used specifications for this parameter
+        # set
+        defaults = dict.fromkeys(pars.keys())
+        options = dict.fromkeys(pars.keys())
+        dtypes = dict.fromkeys(pars.keys())
+        descr = dict.fromkeys(pars.keys())
+
+        # Fill out parameter specifications.  Only the values that are
+        # *not* None (i.e., the ones that are defined) need to be set
+
+        defaults['idsonly'] = False
+        dtypes['idsonly'] = bool
+        descr['idsonly'] = 'Only use the arc lines that have an identified wavelength to trace ' \
+                           'tilts'
+
+        defaults['method'] = 'spline'
+        options['method'] = TraceTiltsPar.valid_methods()
+        dtypes['method'] = str
+        descr['method'] = 'Method used to trace the tilt of the slit along an order.  ' \
+                          'Options are: {0}'.format(', '.join(options['method']))
+
+        # TODO: Need to add checks that check params against method
+        defaults['params'] = [ 1, 1, 0 ]
+        dtypes['params'] = [ int, list ]
+        descr['params'] = 'Parameters to use for the provided method.  TODO: Need more explanation'
+
+        defaults['order'] = 1
+        dtypes['order'] = int
+        descr['order'] = 'Order of the polynomial function to be used for the tilt of an ' \
+                         'individual arc line.  Must be 1 for eschelle data (ARMED pipeline).'
+
+        defaults['disporder'] = 1
+        dtypes['disporder'] = int
+        descr['disporder'] = 'Order of the polynomial function to be used to fit the tilts ' \
+                             'along the dispersion direction.  Only used by ARMED pipeline.'
+
```

Comment:
See WaveTilts for a few more to add in here.

Reply:
Added.  The comments say that method and params are defunct, but I at least found 'method' used somewhere in the code...